src/main/java/com/yelp/nrtsearch/server/luceneserver/field/AtomFieldDef.java

```diff
@@ -145,4 +150,22 @@ public SortField getSortField(SortType type) {
     }
     return sortField;
   }
+
+  @Override
+  public Query getPrefixQuery(
+      PrefixQuery prefixQuery, MultiTermQuery.RewriteMethod rewriteMethod, boolean spanQuery) {
+    verifySearchable("Prefix query");
+    org.apache.lucene.search.PrefixQuery query =
+        new org.apache.lucene.search.PrefixQuery(
+            new Term(prefixQuery.getField(), prefixQuery.getPrefix()));
+    query.setRewriteMethod(rewriteMethod);
+    return query;
+  }
+
+  protected void verifySearchable(String featureName) {
+    if (!isSearchable()) {
+      throw new IllegalStateException(
+          featureName + " requires field to be searchable: " + getName());
+    }
+  }
```

Comment:
It might be better to add this to `IndexableFieldDef` instead of duplicating it in each field